src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -4615,21 +4615,21 @@ checkNullable:
                 value = ParseExpressionCore()
 
             ElseIf modifiers.Any AndAlso modifiers.Any(SyntaxKind.OptionalKeyword) Then
-
-                equals = ReportSyntaxError(InternalSyntaxFactory.MissingPunctuation(SyntaxKind.EqualsToken), ERRID.ERR_ObsoleteOptionalWithoutValue)
-                value = ParseExpressionCore()
-
+                If CheckFeatures(Feature.ImplicitDefaultValueOnOptionalParameter, _scanner.Options) = False Then
+                    equals = ReportSyntaxError(InternalSyntaxFactory.MissingPunctuation(SyntaxKind.EqualsToken), ERRID.ERR_ObsoleteOptionalWithoutValue)
+                    value = ParseExpressionCore()
+                End If
             End If
-
             Dim initializer As EqualsValueSyntax = Nothing
 
-            If value IsNot Nothing Then
+            If (equals IsNot Nothing) AndAlso (value IsNot Nothing) Then
```

Comment:
Given the code `Optional x As Integer, `
In the prior existing code I don't think it gets this far as it an exception.
`Optional parameters must specify a default value.`
In this code it isn't, and the conditional is needed to prevent the null reference. Easier to do the check here rather than modify a syntax node.